polygon/p2p/message_listener.go

```diff
@@ -139,3 +173,47 @@ func (ml *messageListener) notifyInboundMessageObservers(msg *sentry.InboundMess
 		go observer.Notify(msg)
 	}
 }
+
+func (ml *messageListener) listenPeerEvents() {
+	defer ml.stopWg.Done()
+
+	sentrymulticlient.SentryReconnectAndPumpStreamLoop(
+		ml.streamCtx,
+		ml.sentryClient,
+		ml.statusDataFactory(),
+		"PeerEvents",
+		ml.peerEventStreamFactory(),
+		ml.peerEventMessageFactory(),
+		ml.peerEventMessageHandler(),
+		nil,
+		ml.logger,
+	)
+}
+
+func (ml *messageListener) peerEventStreamFactory() sentrymulticlient.SentryMessageStreamFactory {
+	return func(streamCtx context.Context, sentryClient direct.SentryClient) (sentrymulticlient.SentryMessageStream, error) {
+		return sentryClient.PeerEvents(streamCtx, &sentry.PeerEventsRequest{}, grpc.WaitForReady(true))
+	}
+}
+
+func (ml *messageListener) peerEventMessageFactory() sentrymulticlient.MessageFactory[*sentry.PeerEvent] {
+	return func() *sentry.PeerEvent {
+		return new(sentry.PeerEvent)
```

Comment:
please explain this later, I forgot :)

Reply:
`sentrymulticlient.SentryReconnectAndPumpStreamLoop` uses a generic factory to create an empty message (can be different type - either an `InboundMessage` or `PeerEvent`) which is then use to parse raw bytes into.